burr/core/persistence.py

```diff
@@ -142,16 +142,24 @@ class SQLLitePersister(BaseStatePersister):
 
     PARTITION_KEY_DEFAULT = ""
 
-    def __init__(self, db_path: str, table_name: str = "burr_state", serde_kwargs: dict = None):
+    def __init__(
+        self,
+        db_path: str,
+        table_name: str = "burr_state",
+        serde_kwargs: dict = None,
+        allow_multithreading: bool = False,
+    ):
         """Constructor
 
         :param db_path: the path the DB will be stored.
         :param table_name: the table name to store things under.
         :param serde_kwargs: kwargs for state serialization/deserialization.
+        :param allow_multithreading: whether to allow multithreading -- this is good for streamlit, but use at your own risk
```

Comment:
I would rename this to `connect_kwargs` and enable one to pass in everything you can to https://docs.python.org/3/library/sqlite3.html#sqlite3.connect

Reply:
Meh I could go either way, but fine with that

Reply:
At that point we might as well just pass in the connection...

Reply:
> At that point we might as well just pass in the connection...

You could create classmethods for convenience, and then have the constructor take the connection -- but that seems like a bigger change...